### DISCUSSION TITLE
apt-get update failing with "Conflicting distribution" error

### DISCUSSION THREAD

**Discussion by zombiezen** [community/community#25256] [Actions]
<p>I have a GitHub Action workflow that runs an apt-get update. The job’s runs-on is ubuntu-18.04. I get an error like this:</p>
<pre><code>Reading package lists...
W: Conflicting distribution: https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease (expected xenial but got stable)
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Origin' value from 'microsoft-ubuntu-xenial-prod xenial' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Label' value from 'microsoft-ubuntu-xenial-prod xenial' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Suite' value from 'xenial' to 'stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Codename' value from 'xenial' to 'stable'
</code></pre>
<p>This isn’t a repository I added as part of the action, and the workflow worked a couple days ago. Did something change with GitHub Actions that I need to account for?</p>

**Comment by dwsilk** [comment]
<p>Have this problem too, workflow was fine yesterday.</p>

**Comment by anarkiwi** [comment]
<p>/me too</p>
<p>Problem started about 2h ago.</p>

**Comment by medyagh** [comment]
<p>I see the same problem too <a href="https://github.com/kubernetes/minikube/pull/6723/checks?check_run_id=459207826" rel="nofollow noopener">https://github.com/kubernetes/minikube/pull/6723/checks?check_run_id=459207826</a></p>

**Comment by gpritchi-unbounce** [comment]
<p>I also have the same problem</p>
<p>Browsing the package index it’s complaining about, it looks like something was pushed today by microsoft: <a href="https://packages.microsoft.com/ubuntu/16.04/prod/dists/xenial/" rel="nofollow noopener">https://packages.microsoft.com/ubuntu/16.04/prod/dists/xenial/</a></p>

**Comment by obrassard** [comment]
<p>I also face this problem !! </p>

**Comment by ChristopherSt-Pierre** [comment]
<p>Also facing the same issue!</p>

**Comment by Peter-Carr** [comment]
<p>Aslo getting this error, my docker builds are failing :frowning:</p>

**Comment by luigi311** [comment]
<p>Also running into this issue, seems to be a problem with the microsoft repos that are included by default with github actions. This should hopefully be solved soon. It seems to happen every so often with the microsoft repos, last time i had a sync error message from one of their repos and had to rerun the action at a later time.</p>

**Comment by daveball-linz** [comment]
<p>There is an issue with a xenial reference being included in the microsoft-prod list. You can work around it with the following step at the start of any job that involves an apt-get update:</p>
<pre><code>- name: Fix broken apt list
 if: matrix == null || matrix.os == 'ubuntu-latest' || matrix.os == 'ubuntu-18.04' run: sudo perl -p -i -e 's#16\.04/prod xenial#18.04/prod bionic#' /etc/apt/sources.list.d/microsoft-prod.list{,.save}
</code></pre>

**Comment by obrassard** [comment]
<p>The problem seems to be solved, my workflow is now working again! </p>

**Comment by luigi311** [comment]
<p>Looks like it is, my workflows are now completing.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/zombiezen">@zombiezen</a> ,</p>
<p>I noticed that some people replied the  <strong>apt-get update</strong> command has backed to work on their side. I also tested on my side and it also works fine on my side. Please check if it also back to work on your side.</p>
<p>I checked the blog for  <strong><a href="https://www.githubstatus.com/" rel="nofollow noopener">GitHub Status</a></strong> , and found two recent reported service incidents, maybe this issue is related to these two service incidents.</p>
<p>I also have helped you report this issue to the appropriate engineering team, they will evaluate and investigate the issue. If they have any update, I will notify you in time.</p>

**Comment by zombiezen** [comment]
<p>Can confirm this has been fixed. Running my workflow again succeeded. Thank you!</p>

**Comment by mvdan** [comment]
<p>Hm, I’ve restarted my job twice, just now, and I still see errors. Am I the only one?</p>

**Comment by igcocolos** [comment]
<p>using Ubuntu 20.10 and I’ve got the same error. Tried sudo apt-get update: it reads the packages list, but the I am showed the following error:</p>
<p>W: Conflicting distribution: <a href="http://archive.ubuntukylin.com:10006/ubuntukylin" rel="noopener nofollow ugc">http://archive.ubuntukylin.com:10006/ubuntukylin</a> focal InRelease (expected focal but got bionic)</p>
<p>Can someone help please? Thank you!</p>

**Comment by LinqLover** [comment]
<p>FYIO, I experienced the same issue a third time again with Ubuntu 20.04 (latest): <a href="https://github.com/LinqLover/squeak-tracedebugger/runs/5560576131?check_suite_focus=true#step:4:41" class="inline-onebox" rel="noopener nofollow ugc">Fix syntax slip · LinqLover/squeak-tracedebugger@45212e9 · GitHub</a><br>
After ~1h, the issue was resolved by itself, but maybe any hubber out there has an idea how to improve the reliability of these services in the future. :slight_smile:</p>
